### PR TITLE
Migrate riverpod/flutter_riverpod/hooks_riverpod to null-safety

### PR DESCRIPTION
**Pull request by rrousselGit**
- [x] riverpod
  - [x] update the code to use non-nullable types
  - [x] dartdoc
  - [x] README
- [x] flutter_riverpod
  - [x] update the code to use non-nullable types
  - [x] dartdoc
  - [x] README
- [x] hooks_riverpod
  - [x] update the code to use non-nullable types
  - [x] dartdoc
  - [x] README
- [x] examples
  - [x] todos
  - [x] counter
  - [x] stackoverflow
  - [x] marvel
- [x] https://riverpod.dev

### CODE REVIEW

**Comment by gaetschwartz** [comment]
@rrousselGit Do you plan on releasing a prerelease as all code is migrated or waiting on the docs to be ready ?

**Comment by rrousselGit** [comment]
@gaetschwartz I plan on migrating everything today :smile:

**Comment by gaetschwartz** [comment]
Oh even better ! @rrousselGit Will review the docs and try to see if there is anything unclear once you're done ☺️

**Comment by rrousselGit** [comment]
Awesome thanks 😄

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -87,35 +34,30 @@ jobs:
...
       - name: Check format
         run: flutter format --set-exit-if-changed .
-        working-directory: ${{ matrix.package }}
 
       - name: Generate
         run: |
-          if grep -q build_runner "pubspec.yaml"; then
-            flutter pub run build_runner build --delete-conflicting-outputs
-          fi
+          melos exec --scope="${{ matrix.package }}" -- "if grep -q build_runner pubspec.yaml; then dart run build_runner build --delete-conflicting-outputs; fi"
```
Comment (Salakar) [inline .github/workflows/build.yml:50]:
```suggestion
          melos exec --scope="${{ matrix.package }}" --depends-on="build_runner" -- "dart run build_runner build --delete-conflicting-outputs"
```

Would this work?

Comment (rrousselGit) [reply .github/workflows/build.yml]: Thanks :heart:

**Comment by chainchelliah** [comment]
@rrousselGit  I think, you didn't sleep yesterday night and still continuing.
I wonder how it's possible. Have to learn so much from you 🤗😊

**Comment by rrousselGit** [comment]
I'm in an important fight against Github at the moment :devil:

**Comment by chainchelliah** [comment]
Being an Android developer for quite long time I am very much inspired by Jake Wharton.
I would say you're Jake of flutter community. 
(I couldn't resist myself to post this comment 😀)
Gud luck for your fight. Keep fighting 💪

**Comment by gaetschwartz** [comment]
@rrousselGit I've had this issue once, try doing `dart pub cache repair` ?

**Comment by gaetschwartz** [comment]
New error ! Great 😅 If you run `dart pub cache repair` before activating `melos `?

**Comment by rrousselGit** [comment]
Back to the previous error haha

**Comment by gaetschwartz** [comment]
Humm it's weird it looks like melos is messing with something but I'm not familiar enough with it to tell. Maybe @Salakar can help ?

**Comment by venkatd** [comment]
Wow that was a fast migration! Thanks for all the hard efforts on this :)

I don't know if this is of any help, but I did find a GH issue which seemed to have a similar error in CI:
https://github.com/dart-lang/pub/issues/2460

It seems like it could be an issue related to the pub cache path. And apparently need to clear the pub cache manually because `pub cache repair` doesn't work.

Quote below:
> After running rm -rf ~/.pub-cache ; pub get everything works fine again and I can run my tests. Contrary to what I would've expected running pub cache repair didn't fix the issue.

**Comment by rrousselGit** [comment]
It appears that the issue was resolved. Thanks for your help everyone 🥇 

It's getting late. I'll try to publish it tomorrow.

**Comment by rrousselGit** [comment]
Why is `dart test --coverage` generating coverage reports for the test folder and only that? :facepalm:

**Comment by venkatd** [comment]
Swapped our app over to the null safety release, and all works well so far. Great work!

**Pull request merged**